openedx/core/djangoapps/edx_global_analytics/utils.py

```diff
@@ -192,3 +193,96 @@ def platform_coordinates(city_name):
     Module `pytz` also has list of cities with `pytz.all_timezones`.
     """
     return get_coordinates_by_platform_city_name(city_name) or get_coordinates_by_ip()
+
+
+def request_exception_handler_with_logger(function):
+    """
+    Request Exception decorator. Logs error message if it exists.
+    """
+    def request_exception_wrapper(*args, **kwargs):
+        """
+        Decorator wrapper.
+        """
+        try:
+            return function(*args, **kwargs)
+        except requests.RequestException as error:
+            logger.exception(error.message)
+            return
+
+    return request_exception_wrapper
+
+
+def get_access_token():
+    """
+    Return single access token for authorization and dispatch installation statistics to OLGA acceptor.
+    """
+    try:
+        access_token = AccessTokensStorage.objects.first().access_token
+    except AttributeError:
+        access_token = ''
+
+    return access_token
+
+
+@request_exception_handler_with_logger
+def access_token_registration(olga_acceptor_url):
+    """
+    Registry installation on OLGA acceptor via getting access token for further functionality.
+    """
+    token_registration_request = requests.post(olga_acceptor_url + '/api/token/registration/')
+    access_token = token_registration_request.json()['access_token']
+
+    AccessTokensStorage.objects.create(access_token=access_token)
+
+
+@request_exception_handler_with_logger
+def access_token_authorization(olga_acceptor_url):
+    """
+    Verify that installation is allowed access to dispatch installation statistics to OLGA acceptor.
+    """
+    access_token = get_access_token()
+
+    token_authorization_request = requests.post(
+        olga_acceptor_url + '/api/token/authorization/', data={'access_token': access_token, }
+    )
+
+    if token_authorization_request.status_code == 401:
```

Comment:
> why it is needed?

Acceptor can lost `edX instance` token, so `edX instance` won't be authorized.
Solved this problem with refreshed token.